tpp-app/src/home/backend/utils/helpers.js

```diff
@@ -0,0 +1,18 @@
+// Helper functions in the backend used across app
+
+/**
+ * Initializes an empty year array with 12 nested arrays, representing a month.
+ * Within each month array is X null values corresponding to X days of that month in that year.
+ * @param yearNumber number
+ */
+const initializeEmptyYear = (yearNumber) => {
+  let year = new Array(12);
+
+  for (let i = 0; i < 12; ++i) {
+    let daysInMonth = new Date(yearNumber, i+1, 0).getDate();
+    let month = new Array(daysInMonth).fill(null); // fill with daysInMonth null values
```

Comment:
Honestly we don't even need the `.fill(null)`. The new array will be filled with `undefined`. Thoughts?